fighters/common/src/opff/ledges.rs

```diff
@@ -100,8 +101,29 @@ unsafe fn tether_trump_landing(boma: &mut BattleObjectModuleAccessor, status_kin
     */
 }
 
-pub unsafe fn run(boma: &mut BattleObjectModuleAccessor, cat: [i32 ; 4], status_kind: i32, situation_kind: i32, fighter_kind: i32, stick_x: f32, stick_y: f32, facing: f32) {
+// Allow all special moves to grab ledge after their FAF
+// rather than after their animation completes
+unsafe fn global_faf_ledgegrab(fighter: &mut L2CFighterCommon) {
+    // If a special move has a defined FAF (!= 0)
+    // then we allow it to grab ledge
+    if StatusModule::status_kind(fighter.module_accessor) > 0x1DB  // only applies to special moves
+    && fighter.is_situation(*SITUATION_KIND_AIR)
+    && {let motion_kind = MotionModule::motion_kind(fighter.module_accessor);
+        FighterMotionModuleImpl::get_cancel_frame(fighter.module_accessor, Hash40::new_raw(motion_kind), true) > 0.0}
+    {
+        // Both needed to be able to grab ledge
+        if fighter.status_frame() == 1 {
```

Comment:
this implementation will cause gameplay changes for moves that already have `sub_transition_group_check_air_cliff` in its code, allowing them to grab ledge at any point while falling. This should be reimplemented as the following to keep the move's normal ledge grab behavior:
```rs
if cancel_frame > 0.0 {
    if MotionModule::frame == cancel_frame {
        fighter.sub_fighter_cliff_check(GROUND_CLIFF_CHECK_KIND_ON_DROP_BOTH_SIDES.into());
    }
    if is_enable_cancel... {
        sub_transition_group_check_air_cliff
    }
}
```

Reply:
Done, but did not use an == check to prevent any motion rated moves from failing this check